dali/python/nvidia/dali/external_source.py

```diff
@@ -666,20 +680,24 @@ def __call__(
                         kwargs['dtype'] = dtype[i] if i < len(dtype) else nvidia.dali.types.DALIDataType.NO_TYPE
                     else:
                         kwargs['dtype'] = dtype
+                if ndim is not None:
+                    if isinstance(ndim, (list, tuple)):
+                        kwargs['ndim'] = ndim[i] if i < len(ndim) else None
+                    else:
```

Comment:
I feel, that if the operator documentation specifies `ndim` as an int and user provides anything else, I'd just error out. I fear, that allowing 1-element lists or something like that may lead to convoluted type errors. How about going:
```python
if not isinstance(ndim, int):
   raise TypeError("ndim shall be an int")
```

Reply:
This list of ints case is for multiple outputs

Reply:
added info to the docstring